pkg/kubelet/volume/secret/secret.go

```diff
@@ -97,7 +97,7 @@ func (sv *secretVolume) SetUp() error {
 		return fmt.Errorf("Cannot setup secret volume %v because kube client is not configured", sv)
 	}
 
-	secret, err := kubeClient.Secrets(sv.secretRef.Namespace).Get(sv.secretRef.Name)
+	secret, err := kubeClient.Secrets(sv.podRef.Namespace).Get(sv.secretRef.Name)
```

Comment:
Also, this is why you suggested I replace my ClaimRef with ClaimName, which is all Paul and I really need so long as the namespace is available for a lookup within the volume.  We're both looking up things in a namespace by name.   If the namespace is available, it reduces the ref to just name.